docs/how-to-write-function.md

```diff
@@ -0,0 +1,89 @@
+Many of the DLT platforms deal with the issue by having another database such as a RDBMS in front of a ledger to handle application's data in the database and write logs to the ledger as evidence.
+However this scheme is not ideal since data consistency in between the database and the ledger is not always preserved. There could be a case where applications don't have corresponding logs in the ledger due to a catastrophic failure, which defeats the purpose of writing logs to the ledger as evidence.
+Scalar DL resolves the issue with a little differnt approach by introducing Functions to manage applications' data and making Contracts and Functions execute atomically by utilizing underlining distributed ACID transactions.
+
+## Write a Function
+
+Let's take a closer look at `PaymentFunction` to better understand how to write a function.
+
+```java
+public class PaymentFunction extends Function {
+
+  @Override
+  public void invoke(
+      Database database,
+      Optional<JsonObject> functionArgument,
+      JsonObject contractArgument,
+      Optional<JsonObject> contractProperties) {
+    JsonArray array = contractArgument.getJsonArray(ASSETS_ATTRIBUTE_NAME);
+    int amount = contractArgument.getInt(AMOUNT_ATTRIBUTE_NAME);
+    String fromId = array.getString(0);
+    String toId = array.getString(1);
+
+    Optional<Result> account1 = database.get(new Get(new Key(new TextValue("id", fromId))));
+    Optional<Result> account2 = database.get(new Get(new Key(new TextValue("id", toId))));
+
+    long balance1 = ((BigIntValue) account1.get().getValue("balance").get()).get();
+    long balance2 = ((BigIntValue) account2.get().getValue("balance").get()).get();
+
+    // Transfer
+    balance1 -= amount;
+    balance2 += amount;
+
+    Put put1 =
+        new Put(new Key(new TextValue("id", fromId)))
+            .withValue(new BigIntValue("balance", balance1));
+    Put put2 =
+        new Put(new Key(new TextValue("id", toId)))
+            .withValue(new BigIntValue("balance", balance2));
+
+    database.put(put1);
+    database.put(put2);
+  }
+}
+```
+
+As you can easily see, it is a money transfer application, which sends the specified amount of moneny in between two accounts.
+
+### About the arguments
+
+Similar to a Contract using `Ledger` object to manage assets, a Function uses `Database` object to manage records in the underlining database. Note that `Database` implements [Scalar DB](https://github.com/scalar-labs/scalardb) interface so that you can do CRUD operations base on [the data model](https://github.com/scalar-labs/scalardb/blob/master/docs/design.md#data-model) of Scalar DB. 
+
+`JsonObject` for a function argument is a json object and a runtime argument for the function specified by the requester. The argument is not digitally signed as opposed to the contract argument so that it can be used to pass data that is stored in the database but it might be deleted at some later point for some reason.
```

Comment:
```suggestion
`JsonObject` for a function argument is a JSON object and a runtime argument for the function specified by the requester. The argument is not digitally signed as opposed to the contract argument so that it can be used to pass data that is stored in the database but it might be deleted at some later point for some reason.
```